Bugzilla/WebService/Bugzilla.pm

```diff
@@ -80,6 +83,36 @@ sub time {
     };
 }
 
+sub jobqueue_status {
+    my ( $self, $params ) = @_;
+    
+    Bugzilla->login(LOGIN_REQUIRED);
+
+    my $dbh = Bugzilla->dbh;
+    my $query = qq{
+        SELECT
+            COUNT(*) AS total,
+            COALESCE(
+                (SELECT COUNT(*)
+                    FROM ts_error
+                    WHERE ts_error.jobid = j.jobid
+                ) 
+            , 0) AS errors
+        FROM ts_job j
+            INNER JOIN ts_funcmap f
+                ON f.funcid = j.funcid;
+    };
+
+    my $status = {};
+    try {
+        $status = $dbh->selectall_arrayref($query, { Slice => {} });
+    } catch {
+        ThrowCodeError( "jobqueue_status_failed", { message => "Unable to report job queue status at this time." } );
```

Comment:
- Is there a better error code to use? Here I'm using a string instead of one of the error codes defined in Bugzilla::Constants because I couldn't find one that fit.
- This ThrowCodeError doesn't seem to work as expected. When I trigger an error with an invalid SQL query, I end up getting `{}` in the response instead of the normal json error response. Could I be doing something wrong here?

Reply:
Actually I see now that this error code is causing a test failure.

Reply:
This interesting! I have no idea what is going on.

Reply:
Perhaps `jobqueue_status_error`.
Also `use Bugzilla::Logging` and use ERROR($_); to log the error.